ops/model.py

```diff
@@ -1865,8 +1865,8 @@ class MaintenanceStatus(StatusBase):
 class WaitingStatus(StatusBase):
     """A unit is unable to progress.
 
-    The unit is unable to progress to an active state because an application to which
-    it is related is not running.
+    The unit is unable to progress to an active state because an application with which
+    it is integrated is not running.
```

Comment:
@benhoyt this seems very specific about the use-case for `WaitingStatus`. Isn't it ok to use it if you're waiting on *anything*, rather than specifically for an integration/relation?

Reply:
That's taken straight from the comment in Juju [core/status/status.go](https://github.com/juju/juju/blob/55fb5d03683b70dea08d1e54687e0f90bb06a63a/core/status/status.go#L143-L146), so I presume it's intentional:

```
 	// The unit is unable to progress to an active state because an application to
	// which it is related is not running.
```

I think you'd use MaintenanceStatus otherwise.

Reply:
> That's taken straight from the comment in Juju [core/status/status.go](https://github.com/juju/juju/blob/55fb5d03683b70dea08d1e54687e0f90bb06a63a/core/status/status.go#L143-L146), so I presume it's intentional:

Ah, ok. It should presumably stay in that case.

> I think you'd use MaintenanceStatus otherwise.

Hmm, if I'm waiting for the container to be ready or waiting for an API to come up, `WaitingStatus` seems the natural one to use. `MaintenanceStatus` seems like it's more when the charm is actively doing something.